dom-fetch-client/src/main/scala/org/http4s/dom/FetchClient.scala

```diff
@@ -20,30 +20,71 @@ package dom
 import cats.effect.Async
 import cats.effect.Resource
 import cats.syntax.all._
+import cats.effect.syntax.all._
+import java.util.concurrent.TimeoutException
 import org.http4s.client.Client
+import org.http4s.headers.{`Referer`}
 import org.scalajs.dom.crypto._
+import org.scalajs.dom.experimental.AbortController
 import org.scalajs.dom.experimental.Fetch
 import org.scalajs.dom.experimental.Headers
 import org.scalajs.dom.experimental.HttpMethod
 import org.scalajs.dom.experimental.RequestInit
+import org.scalajs.dom.experimental.{Response => FetchResponse}
+import scala.concurrent.duration._
 
 object FetchClient {
 
-  def apply[F[_]](implicit F: Async[F]): Client[F] = Client[F] { req: Request[F] =>
+  private[dom] def makeClient[F[_]](
```

Comment:
Maybe we make `FetchClient` to `private[dom]`?

Reply:
I followed the pattern from `BlazeClient` here, but yes, we can make it private. It just encapsulates the builder method.